core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -712,7 +712,8 @@ object KafkaConfig {
     "If it is not set, the metadata log is placed in the first log directory from log.dirs."
   val MetadataSnapshotMaxNewRecordBytesDoc = "This is the maximum number of bytes in the log between the latest snapshot and the high-watermark needed before generating a new snapshot."
   val ControllerListenerNamesDoc = "A comma-separated list of the names of the listeners used by the controller. This is required " +
-    "if running in KRaft mode. The ZK-based controller will not use this configuration."
+    "if running in KRaft mode. When communicating with the controller quorum, the broker will always use the first listener in this list.\n " +
+    "Note: The ZK-based controller should not set this configuration."
```

Comment:
After https://github.com/apache/kafka/pull/11503, a ZooKeeper-based cluster that sets this config will fail to restart until this config is removed. We should note it in the doc, not just "ignore the config" as before. Besides, we should also have simple description explain we will use the 1st listener to communicating with the controller quorum. (ref: [KIP-631](https://cwiki.apache.org/confluence/display/KAFKA/KIP-631%3A+The+Quorum-based+Kafka+Controller))

Before
```
A comma-separated list of the names of the listeners used by the controller. This is required if running in KRaft mode. The ZK-based controller will not use this configuration.
```

After:
```
A comma-separated list of the names of the listeners used by the controller. This is required if running in KRaft mode. When communicating with the controller quorum, the broker will always use the first listener in this list. 
Note: The ZK-based controller should not set this configuration. 
```